src/panels/lovelace/hui-header.ts

```diff
@@ -0,0 +1,750 @@
+        overflowAction: this._handleUnusedEntities,
+        visible: !this._editMode && this._yamlMode,
+        overflow: true,
+      },
+      {
+        icon: mdiRefresh,
+        key: "ui.panel.lovelace.menu.reload_resources",
+        overflowAction: this._handleReloadResources,
+        visible:
+          !this._editMode &&
+          (this.hass.panels.lovelace?.config as LovelacePanelConfig)?.mode ===
+            "yaml",
+        overflow: true,
+      },
+      {
+        icon: mdiPencil,
+        key: "ui.panel.lovelace.menu.configure_ui",
+        overflowAction: this._handleEnableEditMode,
+        buttonAction: this._enableEditMode,
+        visible:
+          !this._editMode &&
+          this.hass!.user?.is_admin &&
+          !this.hass!.config.recovery_mode,
+        overflow: true,
+        overflow_can_promote: true,
+      },
+    ];
+
+    const overflowItems = items.filter((i) => i.visible && i.overflow);
+    const overflowCanPromote =
+      overflowItems.length === 1 && overflowItems[0].overflow_can_promote;
+    const buttonItems = items.filter(
+      (i) => i.visible && (!i.overflow || overflowCanPromote)
+    );
+
+    buttonItems.forEach((i) => {
+      result.push(
+        html`<ha-tooltip
+          slot="actionItems"
+          placement="bottom"
+          .content=${[this.hass!.localize(i.key), i.suffix].join(" ")}
+        >
+          <ha-icon-button
+            .path=${i.icon}
+            @click=${i.buttonAction}
+          ></ha-icon-button>
+        </ha-tooltip>`
+      );
+    });
+    if (overflowItems.length && !overflowCanPromote) {
+      const listItems: TemplateResult[] = [];
+      overflowItems.forEach((i) => {
+        listItems.push(
+          html`<ha-list-item
+            graphic="icon"
+            @request-selected=${i.overflowAction}
+          >
+            ${[this.hass!.localize(i.key), i.suffix].join(" ")}
+            <ha-svg-icon slot="graphic" .path=${i.icon}></ha-svg-icon>
+          </ha-list-item>`
+        );
+      });
+      result.push(
+        html`<ha-button-menu slot="actionItems">
+          <ha-icon-button
+            slot="trigger"
+            .label=${this.hass!.localize("ui.panel.lovelace.editor.menu.open")}
+            .path=${mdiDotsVertical}
+          ></ha-icon-button>
+          ${listItems}
+        </ha-button-menu>`
+      );
+    }
+    return html`${result}`;
+  }
+
+  protected render(): TemplateResult {
+    const views = this.lovelace?.config.views ?? [];
+    const curViewConfig =
+      typeof this.curView === "number" ? views[this.curView] : undefined;
+    const dashboardTitle = this.panel
+      ? getPanelTitle(this.hass, this.panel)
+      : undefined;
+
+    const _isTabHiddenForUser = (view: LovelaceViewConfig) =>
+      view.visible !== undefined &&
+      ((Array.isArray(view.visible) &&
+        !view.visible.some((e) => e.user === this.hass!.user?.id)) ||
+        view.visible === false);
+
+    const visibleViews = views.filter(
```

Comment:
We probably don't want to calculate this here, but in `hui-lovelace-navigation`?